telegram/ext/_handlers/callbackqueryhandler.py

```diff
@@ -139,22 +151,39 @@ def check_update(self, update: object) -> Optional[Union[bool, object]]:
 
         """
         # pylint: disable=too-many-return-statements
-        if isinstance(update, Update) and update.callback_query:
-            callback_data = update.callback_query.data
-            if self.pattern:
-                if callback_data is None:
-                    return False
-                if isinstance(self.pattern, type):
-                    return isinstance(callback_data, self.pattern)
-                if callable(self.pattern):
-                    return self.pattern(callback_data)
-                if not isinstance(callback_data, str):
-                    return False
-                if match := re.match(self.pattern, callback_data):
-                    return match
-            else:
-                return True
-        return None
+        if not (isinstance(update, Update) and update.callback_query):
+            return None
+
+        callback_data = update.callback_query.data
+        game_short_name = update.callback_query.game_short_name
+
+        if not any([self.pattern, self.game_pattern]):
+            return True
+
+        # we check for .data or .game_short_name from update to filter based on whats coming
+        # this gives xor-like behavior
+        if callback_data:
+            if not self.pattern:
+                return False
+            if isinstance(self.pattern, type):
+                return isinstance(callback_data, self.pattern)
+            if callable(self.pattern):
+                return self.pattern(callback_data)
+            if not isinstance(callback_data, str):
+                return False
+            if match := re.match(self.pattern, callback_data):
+                return match
+
+        elif game_short_name:
+            if not self.game_pattern:
+                return False
+            if not isinstance(game_short_name, str):
+                return False
```

Comment:
IISC this can be removed - line 177 already takes care of it.

Reply:
Removing that code would break tests [here #L144](https://github.com/jainamoswal/python-telegram-bot/blob/21f4b3b568c3d6b8961d2a4bb169cde04841e077/tests/ext/test_callbackqueryhandler.py#L144) 

Or do you want me to remove that code from test files (as the `.data` is never going to be an object? 🤔)